types/parser_driver/value_expr.go

```diff
@@ -69,11 +68,45 @@ type ValueExpr struct {
 	projectionOffset int
 }
 
-// Restore implements Recoverable interface.
-func (n *ValueExpr) Restore(sb *strings.Builder) error {
-	err := n.format(sb)
-	if err != nil {
-		return errors.Trace(err)
+// Restore implements Node interface.
+func (n *ValueExpr) Restore(ctx *ast.RestoreCtx) error {
+	switch n.Kind() {
+	case types.KindNull:
+		ctx.WriteKeyWord("NULL")
+	case types.KindInt64:
+		if n.Type.Flag&mysql.IsBooleanFlag != 0 {
```

Comment:
what is `mysql.IsBooleanFlag` ? i copy it from format.
```
if n.Type.Flag&mysql.IsBooleanFlag != 0 {
			if n.GetInt64() > 0 {
				ctx.WriteKeyWord("TRUE")
			} else {
				ctx.WriteKeyWord("FALSE")
			}
}
```
is it right?

Reply:
It's right. boolean is represent in integer.